Assets/Scripts/Configuration/ConfiguratorStarter.cs

```diff
@@ -7,8 +8,11 @@ public class ConfiguratorStarter : MonoBehaviour
         [SerializeField]
         private Configurator configurator;
 
-        private void Awake()
```

Comment:
This was needed because the WebRequest triggered through the DossierLoader doesn't work in Awake. It makes more sense anyway to load the config a frame after the application initialized

Reply:
Ik heb initieel Awake gebruikt omdat ik bezig was om de RDOrigin handmatig van een andere positie te voorzien; en aangezien die bepaalde initialisatie maar 1x doet in Start dacht ik: chips, dat moet in Awake. Dat principe heb ik alleen losgelaten en sindsdien stond dit nog in Awake.

Start is de meest logische plek; maar je kan met volgordelijkheid zitten omdat je de Configuratie eigenlijk wil toepassen op MonoBehaviours die al hun Start hebben gehad, op die manier weet je zeker dat de initialisatie van die monobehaviours gebeurd is voordat je middels configuratie die gaat manipuleren.

De andere oplossing zou zijn geweest om in de project settings de Script Execution Order aan te passen -maar dat leek mij niet onderhoudsvriendelijk en foutgevoelig- of de DefaultExecutionOrder attribute te gebruiken -beetje zelfde probleem, maar in ieder geval regel je het niet in de project settings.

Dus, bovenstaande bij elkaar opgeteld leek het mij verstandig om het toepassen van de configuratie een frame later te doen. De enige caveat die ik wel kan bedenken is als iemand in de Start methode de configuratie zou willen uitlezen. Maar dat lijkt me niet handig vanuit een Package manier van denken (kom ik hieronder nog even op terug in de DossierListener feedback), dat is ook de reden waarom ik scheutig ben met het toevoegen van UnityEvents op objecten die een soort van lifecycle hebben

Reply:
Volgens mij beschrijf je precies het probleem waar ik het over had met Awake/Start. De RDOrigin moet je eigenlijk niet willen aanpassen in Awake, aangezien je niet weet of dit dan voor de RDOrigin initialisatie gebeurt of erna. Je moet in Start geen dingen willen doen die afhankelijk zijn van de execution van andere Start functies, omdat je niet weet op welke volgorde dat gebeurt. Wat je nu doet werkt daar omheen, maar als dit in meerdere scripts gaat gebeuren dan krijg je weer dat je niet weet of de IEnumerator Start voor of na een andere IEnumerator Start gebeurt. Als het afhankelijk is van initialisatie van een ander script zou dit in void Start() moeten gebeuren en de initialisatie van een ander script in void Awake(). Als er echt problemen zijn met dingen die heen en weer op elkaar moeten wachten zou ik een coroutine maken met WaitUntil() zodat je niet in het project functies hebben die een frame wachten en waarvan iemand die niet bekend is met de code niet van weet waarom dat zo is.

Script execution order zou ik ook niet aan gaan zitten, dat is inderdaad erg onderhouds-/foutgevoelig.

Reply:
Zoals besproken, laten we eerst doorspreken wat we willen; en of we niet meer aandacht aan een plugin systeem moeten besteden of uberhaupt even nadenken hoe de Configuratie in het geheel past. Voor nu heb ik de constructie eruit gehaald; ik denk dat dat kan.